### PR TITLE
OpcodeDispatcher: Implement MULX

### PR DESCRIPTION
**Pull request by lioncash**
Now with all the other instructions out of the way, we have the fun ones left to implement (PDEP and PEXT)

### CODE REVIEW

File: External/FEXCore/Source/Interface/Core/OpcodeDispatcher.cpp [hunk 1]
```diff
@@ -2372,6 +2372,29 @@ void OpDispatchBuilder::RORX(OpcodeArgs) {
   StoreResult(GPRClass, Op, Result, -1);
 }
 
+void OpDispatchBuilder::MULX(OpcodeArgs) {
+  // RDX is the implied source operand in the instruction
+  const auto RDXOffset = offsetof(FEXCore::Core::CPUState, gregs[FEXCore::X86State::REG_RDX]);
+  const auto OperandSize = GetSrcSize(Op);
+
+  // Intel manual states that operands are always 32-bit
```
Comment (Sonicadvance1) [inline External/FEXCore/Source/Interface/Core/OpcodeDispatcher.cpp:2380]: I believe this is incorrect. It should be that VEX.W is ignored. So 32-bit is always 32-bit operation.
Encoding the assembly manually for a 32-bit ASM test would prove this.

**Comment by Sonicadvance1** [comment]
This reminds me that I should spend some time getting a 32-bit host runner hooked up to CI for the ASM tests.
This way we can easily test things like this.

File: External/FEXCore/Source/Interface/Core/OpcodeDispatcher.cpp (hunk 1 above)
Comment (lioncash) [reply External/FEXCore/Source/Interface/Core/OpcodeDispatcher.cpp]: Amended to remove the decode failure

Comment (Sonicadvance1) [reply External/FEXCore/Source/Interface/Core/OpcodeDispatcher.cpp]: With the decode failure removed. What is the current behaviour of VEX.W in 32-bit?
Does FEX currently debug this as a 64-bit op incorrectly?

Comment (lioncash) [reply External/FEXCore/Source/Interface/Core/OpcodeDispatcher.cpp]: Not anymore =)

**Review by Sonicadvance1 — APPROVED** [review]

**Pull request merged**